### PR TITLE
Support relative toolchain filepath

### PR DESCRIPTION
**Pull request by Ubiquite**
forwarding absolute path from CMAKE_TOOLCHAIN_FILE since hunter build directory isn't the same as user's

### CODE REVIEW

File: cmake/schemes/url_sha1_cmake.cmake.in [hunk 1]
```diff
@@ -112,6 +112,8 @@ foreach(configuration @HUNTER_PACKAGE_CONFIGURATION_TYPES@)
 
   set(current_project "@HUNTER_EP_NAME@-${configuration}")
 
+  get_filename_component(absolute_CMAKE_TOOLCHAIN_FILE "${CMAKE_TOOLCHAIN_FILE}" ABSOLUTE)
```
Comment (ruslo) [inline cmake/schemes/url_sha1_cmake.cmake.in]: This code is running with toolchain created by Hunter:
* https://github.com/ruslo/hunter/blob/382aec18eba519d49c34491106a8e125cbc035e0/cmake/modules/hunter_download.cmake#L243

It should be absolute already.

File: scripts/create-toolchain-info.cmake [hunk 2]
```diff
@@ -64,11 +64,13 @@ if(NOT EXISTS "${predefined}")
   hunter_internal_error("${predefined} not exists")
 endif()
 
+get_filename_component(absolute_CMAKE_TOOLCHAIN_FILE "${CMAKE_TOOLCHAIN_FILE}" ABSOLUTE)
```
Comment (ruslo) [inline scripts/create-toolchain-info.cmake]: You already use absolute path in `hunter_calculate_toolchain_sha1.cmake`, so this one not needed also (?).

**Comment by Ubiquite** [comment]
Okey I remove the changes on these two files

**Comment by ruslo** [comment]
Have you actually test it? Doesn't seem to work, I think HunterGate should be updated too.

**Comment by Ubiquite** [comment]
nop 
I'm testing now with modified HunterGate

**Comment by ruslo** [comment]
> with modified HunterGate

You should open pull request to HunterGate too. This change make no sense without HunterGate update.

**Comment by Ubiquite** [comment]
just editing the HunterGate seems to fix our build, I created a PR : https://github.com/hunter-packages/gate/pull/15

**Comment by ruslo** [comment]
> I created a PR : hunter-packages/gate#15

Merged

> just editing the HunterGate seems to fix our build

Same for me. Closing?

**Pull request closed without merging**